src/libraries/System.Drawing.Common/tests/BitmapTests.cs

```diff
@@ -1750,22 +1750,53 @@ public void FromNonSeekableStream()
             }
         }
 
-        private class NonSeekableStream : Stream
+        [ConditionalTheory(Helpers.IsDrawingSupported)]
+        [InlineData(false, true)]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        public void SaveToRestrictiveStream(bool disableRead, bool disableSeek)
+        {
+            using (Stream backingStream = new MemoryStream())
+            using (Stream restrictiveStream = new TestStream(backingStream, disableRead, disableSeek))
+            { 
+                using (Bitmap bitmap = new Bitmap(100, 100))
+                {
+                    bitmap.Save(restrictiveStream, ImageFormat.Png);
+                }
+
+                using (Bitmap bitmap = new Bitmap(backingStream))
+                {
+                    Assert.Equal(100, bitmap.Height);
+                    Assert.Equal(100, bitmap.Width);
+                    Assert.Equal(ImageFormat.Png, bitmap.RawFormat);
+                }
+            }
+        }
+
+        private class TestStream : Stream
         {
             private Stream _stream;
+            private bool _disableRead;
+            private bool _disableSeek;
 
-            public NonSeekableStream(Stream stream)
+            public TestStream(Stream stream, bool disableRead = false, bool disableSeek = false)
             {
                 _stream = stream;
+                _disableRead = disableRead;
+                _disableSeek = disableSeek;
             }
 
-            public override bool CanRead => _stream.CanRead;
-            public override bool CanSeek => false;
+            public override bool CanRead => !_disableRead && _stream.CanRead;
+            public override bool CanSeek => !_disableSeek && _stream.CanSeek;
             public override bool CanWrite => _stream.CanWrite;
             public override long Length => _stream.Length;
-            public override long Position { get => _stream.Position; set => throw new InvalidOperationException(); }
+            public override long Position
+            {
+                get => _stream.Position;
+                set => _stream.Position = _disableSeek ? throw new InvalidOperationException() : value;
+            }
             public override void Flush() => _stream.Flush();
-            public override int Read(byte[] buffer, int offset, int count) => _stream.Read(buffer, offset, count);
+            public override int Read(byte[] buffer, int offset, int count) => _disableRead ? throw new InvalidOperationException() : _stream.Read(buffer, offset, count);
```

Comment:
Nit: streams generally throw NotSupportedException in such cases

Reply:
I was following what the stream already here did, but I can change that.